res/controllers/Traktor-Kontrol-S4-MK2-hid-scripts.js

```diff
@@ -1500,3 +1531,32 @@ TraktorS4MK2.onLoopEnabledChanged = function(value, group, key) {
   TraktorS4MK2.outputCallbackLoop(value, group, "loop_in");
   TraktorS4MK2.outputCallbackLoop(value, group, "loop_out");
 }
+
+TraktorS4MK2.onLoopSizeChanged = function(value, group, key) {
+  var deck = TraktorS4MK2.resolveDeckIfActive(group);
+  //deal with single digit values
+  if (value.toString().length === 1){
+    TraktorS4MK2.sendLoopSizeMessage(deck, '', value, false, false);
+  // values with two digits
+  } else if (value.toString().length === 2 ) {
+    TraktorS4MK2.sendLoopSizeMessage(deck, value.toString().split("")[0], value.toString().split("")[1], false, false);
+  // deal with fraction beats
+  } else if (1 > value > 0 ) {
+    if (value === 0.5){
+      TraktorS4MK2.sendLoopSizeMessage(deck, '-', 2, false, false);
+    } else if (value === 0.25){
+      TraktorS4MK2.sendLoopSizeMessage(deck, '-', 4, false, false);
+    } else if (value === 0.125){
+      TraktorS4MK2.sendLoopSizeMessage(deck, '-', 8, false, false);
+    } else {
+      TraktorS4MK2.sendLoopSizeMessage(deck, '-', 'n', false, false);
```

Comment:
It's great that you got this working! My main motivation for adding beatloop_size in #1187 was to support controllers like this, so I'm really glad this can be in the same Mixxx release.

If you use the dot instead of `-` to indicate a fraction, you could use both digits of the display for fractional sizes. This is how it is done on the Hercules P32. The code could also be more robust by showing the reciprocal of the loop size (1 / value) instead of checking for specific fractions. How does Traktor show fractional loop sizes?

Reply:
I'm not sure about Traktor, I haven't actually used it except for with the S2 which doesn't have a loop counter.

Noted - although not sure how to make the code more robust with the fractions.

Reply:
Try checking if `1/value` is an integer. If it is, show that on the LEDs with a dot.